### PR TITLE
Add test plan for shopping cart

### PR DESCRIPTION
**Pull request by arrubin**
Based on initial requirements, I have developed an initial test plan for the
shopping cart.

### CODE REVIEW

File: shopping_cart/features/authentication.feature [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Feature: authentication
+
+  As a customer
```
Comment (jvanderz22) [inline shopping_cart/features/authentication.feature]: I think you're missing some scenarios relating to login here. What happens if the username doesn't exist? Or the password is wrong?

File: shopping_cart/features/update_cart.feature [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Feature: update shopping cart
+
+  As a customer
```
Comment (jvanderz22) [inline shopping_cart/features/update_cart.feature]: These scenarios look good, but I think there is room for a little wider coverage here. Are we sure that nothing happens to the other items in the cart if we make a change to one of them?

**Comment by jvanderz22** [comment]
There's room for a more expansive test plan to be implemented here, but you do a good job covering the basic scenarios, including happy, sad, and bad paths. The features and step definitions are given descriptive names that make sense in plain english and all of the scenarios follow good cucumber syntax. 

Review the areas where additional tests would provide greater code coverage, but you don't have to resubmit this exercise.

File: shopping_cart/features/authentication.feature [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    Given my cart contains items
+    And I am not authenticated
+    When I login
+    Then the items appear in my cart
+
+  Scenario: log out with items in cart
+    Given my cart contains items
+    And I am authenticated
+    When I login
+    Then the items appear in my cart
```
Comment (djkotowski) [inline shopping_cart/features/authentication.feature]: What about merging carts?  (Acct has 1 item X and 1 item Y, anon user has 1 item X in cart; when anon user logs in to acct, cart has 2 item X and 1 item Y).

**Comment by djkotowski** [comment]
Ditto what Jordan said.  Take a look at the comments, but no need for resubmission.
